contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -428,11 +448,18 @@ transform_from_ci_as_for_likenode(Node *node, OpExpr *op, like_ilike_info_t like
 		Node	   *constant_suffix;
 		Const	   *highest_sort_key;
 
+		/* Always create a CollateExpr on top to match with op->inputcollid */
+		new_leftop->arg = linitial(op->args);
+		new_leftop->collOid = op->inputcollid;
+		linitial(op->args) = (Node*) new_leftop;
+
 		/* construct leftop >= pattern */
-		optup = compatible_oper(NULL, list_make1(makeString(">=")), ltypeId, ltypeId,
+		optup = compatible_oper(NULL, list_make1(makeString(">=")), ltypeId, rtypeId,
```

Comment:
What's the purpose of change ltypeId to rtypeId in here ?

Reply:
Can you add an assert that those two are the same in here ?

Reply:
Added Assert. Also, want to correct my previous statement, the operands can be of different datatypes like text LIKE name. However, for the optimisation part, both of them would be TEXT